pkg/vault/write.go

```diff
@@ -0,0 +1,24 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+)
+
+func (vc *vaultClient) write(engine, key, value string) (*api.Secret, error) {
+	vc.tracer.trace(fmt.Sprintf("%s/write", vc.config.tracePrefix))
+
+	data := map[string]interface{}{
+		"data": map[string]interface{}{
+			key: value,
+		},
+	}
+
+	secret, err := vc.client.Logical().Write(engine, data)
```

Comment:
It appears that executing write here will remove any secret not included in the data, it's not an append action.

Working on a fox for this.